src/nodedef.cpp

```diff
@@ -112,6 +183,71 @@ void NodeBox::deSerialize(std::istream &is)
 		wall_side.MinEdge = readV3F1000(is);
 		wall_side.MaxEdge = readV3F1000(is);
 	}
+	else if (type == NODEBOX_CONNECTED)
+	{
+		u16 count = readU16(is);
+		fixed.reserve(count);
+		while (count--) {
+			aabb3f box;
```

Comment:
Actually, that totally doesn't work.

I think the problem is that the compiler reorders the two reads when it creates the new aabb3f in the constructor, since with this change all the collision boxes are broken, at least that's my guess (somehow they are drawn correctly, but I can move through them).

I've eliminated all the other changes from this particular problem, it's really the compound push_back(aabb3f()) here that does not work.